Mathlib/GroupTheory/Order/Min.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.GroupTheory.Torsion
+
+/-!
+# Minimum order of an element
+
+This file defines the minimum order of an element of a monoid.
+
+## Main declarations
+
+* `Monoid.minOrder`: The minimum order of an element of a given monoid.
+* `Monoid.minOrder_eq_top`: The minimum order is infinite iff the monoid is torsion-free.
+* `ZMod.minOrder`: The minimum order of $$ℤ/nℤ$$ is the smallest factor of `n`, unless `n = 0, 1`.
+-/
+
+open Subgroup
+
+variable {α : Type*}
+
+namespace Monoid
+section Monoid
+variable (α) [Monoid α]
+
+/-- The minimum order of a non-identity element. Also the minimum size of a nontrivial subgroup.
+Returns `∞` if the monoid is torsion-free. -/
+@[to_additive "The minimum order of a non-identity element. Also the minimum size of a nontrivial
+subgroup. Returns `∞` if the monoid is torsion-free."]
+noncomputable def minOrder : ℕ∞ := ⨅ (a : α) (_ha : a ≠ 1) (_ha' : IsOfFinOrder a), orderOf a
+
+variable {α} {a : α}
+
+@[to_additive (attr := simp)]
+lemma minOrder_eq_top : minOrder α = ⊤ ↔ IsTorsionFree α := by simp [minOrder, IsTorsionFree]
+
+@[to_additive (attr := simp)] protected alias ⟨_, IsTorsionFree.minOrder⟩ := minOrder_eq_top
+
+@[to_additive (attr := simp)]
+lemma le_minOrder {n : ℕ∞} :
+    n ≤ minOrder α ↔ ∀ ⦃a : α⦄, a ≠ 1 → IsOfFinOrder a → n ≤ orderOf a := by simp [minOrder]
+
+@[to_additive]
+lemma minOrder_le_orderOf (ha : a ≠ 1) (ha' : IsOfFinOrder a) : minOrder α ≤ orderOf a :=
+  le_minOrder.1 le_rfl ha ha'
+
+end Monoid
+
+variable [Group α] {s : Subgroup α} {n : ℕ}
+
+@[to_additive]
+lemma le_minOrder' {n : ℕ∞} :
```

Comment:
I think this one can have a name mentioning subgroup in some way, rather than `'`. Also let's reference this lemma in the docstring of `minOrder`, since you quote this fact.